Makefile.am

```diff
@@ -212,13 +199,19 @@ tools/debian/copyright: tools/debian/copyright.template $(MANIFESTS)
 	$(AM_V_GEN) $(srcdir)/tools/build-debian-copyright > $@.tmp
 	$(MV) $@.tmp $@
 
-# Build up the distribution using $COMMITTED_DIST and include node_modules licenses
-# also automatically update minimum base dependency in RPM spec file
-dist-hook:: $(MANIFESTS)
+distdir: $(WEBPACK_INPUTS) $(WEBPACK_OUTPUTS) $(WEBPACK_TEST_DEPS) $(MANIFESTS)
+if !ENABLE_DOC
+	@echo "*** doc must be enabled (ie: --enable-doc) in order to make dist or distcheck"
+	@false
+endif
+	$(MAKE) $(AM_MAKEFLAGS) distdir-am
```

Comment:
Random comment from the peanut gallery: I find `dist-hook::` much more elegant than a recursive `make` call with a magic internal target. However, I do agree to merging all the `dist-hook::`s into one.

Reply:
Funny: the magical make invocation with the internal target needs to stay, but `dist-hook:` is back anyway, specifically because we need the recursive make invocation to get the correct values of `$(WEBPACK_INPUTS)` and so after the build completes.